bennu-portal/src/main/java/org/fenixedu/bennu/portal/domain/MenuContainer.java

```diff
@@ -360,24 +353,57 @@ public void setPath(String path) {
         }
     }
 
+    @Override
+    public Set<com.qubit.terra.portal.domain.functionalities.Application> getApplications() {
+        String availableApplicationNames = getAvailableApplicationNames();
+        return availableApplicationNames == null ? Collections.emptySet() : Stream.of(availableApplicationNames.split(","))
+                .map(key -> ApplicationRegistry.getAppByKey(key))
+                .filter(app -> app != null && app.getFunctionalities().size() > 0).collect(Collectors.toSet());
+    }
+
+    @Override
+    public List<com.qubit.terra.portal.domain.menus.MenuItem> getChilds() {
+        return getOrderedChild().stream().map(t -> (MenuItem) t).collect(Collectors.toList());
+    }
+
+    @Override
+    public void addChildMenuItem(com.qubit.terra.portal.domain.menus.MenuItem menuItem) {
+        addChild((MenuItem) menuItem);
+    }
+
+    @Override
+    public boolean isRootApplicationMenu() {
+        return isRoot();
+    }
+
+    @Override
     public void updateAccessGroup() {
         String groupExpression =
                 getChildSet().stream().map(item -> item.getAccessGroup().getExpression()).collect(Collectors.joining(" | "));
         setAccessGroup(Group.parse(groupExpression));
     }
 
-    public Set<Application> getApplications() {
-        String availableApplicationNames = getAvailableApplicationNames();
-        return availableApplicationNames == null ? Collections.emptySet() : Stream.of(availableApplicationNames.split(","))
-                .map(key -> ApplicationRegistry.getAppByKey(key))
-                .filter(app -> app != null && app.getFunctionalities().size() > 0).collect(Collectors.toSet());
+    @Override
+    public Set<com.qubit.terra.portal.domain.functionalities.Application> availableApplicationKeys() {
```

Comment:
@joaofreitas-qubit this method should be extracted to the MenuService implementation for fenix